pkg/rpc/response/result/invoke.go

```diff
@@ -94,36 +108,36 @@ func (r *Invoke) Finalize() {
 
 // MarshalJSON implements the json.Marshaler.
 func (r Invoke) MarshalJSON() ([]byte, error) {
-	defer r.Finalize()
 	var (
-		st       json.RawMessage
-		err      error
-		faultSep string
-		arr      = make([]json.RawMessage, len(r.Stack))
+		st              json.RawMessage
+		err             error
+		faultSep        string
+		arr             = make([]json.RawMessage, len(r.Stack))
+		sessionsEnabled = r.onNewSession != nil
+		sessionID       string
+		iterators       []ServerIterator
 	)
 	if len(r.FaultException) != 0 {
 		faultSep = " / "
 	}
-arrloop:
 	for i := range arr {
 		var data []byte
 		if (r.Stack[i].Type() == stackitem.InteropT) && iterator.IsIterator(r.Stack[i]) {
-			iteratorValues, truncated := iterator.Values(r.Stack[i], r.maxIteratorResultItems)
```

Comment:
One thing I don't like about the C# patch (and other users are complaining too) is that it breaks backwards compatibility, existing apps just break after this change. What I would like to get is for the system to work exactly as it worked before (in 0.99.0) with (the default) `SessionEnabled=false`. It will also make `InvokeAndPackIteratorResults` less relevant (although there still might be some use of it).

Reply:
Yes, I also don't like it and I've left a comment about it in the C# PR (https://github.com/neo-project/neo-modules/pull/715#discussion_r897641254), but seems that it's done intentionally:
```
I think public RPC nodes should only expose part of the API
```
I'm afraid, if we're about to keep an old behaviour, then the compatibility will be lost.

Reply:
We can change it later, but IMO keeping compatibility with ourselves (0.99.0) is a priority here. Also, public RPCs just can't have `SessionEnabled=true` now (it's trivial to DoS them), so it makes them useless for any requests returning iterators (you get an ID that you can't use in any way). Keeping the old behavior will cover more cases IMO.

Reply:
Fixed, now if sessions are disabled then up to `MaxIteratorResultItems` is returned.